packages/cadl-ranch-specs/http/type/model/templated/main.tsp

```diff
@@ -0,0 +1,62 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Illustrates the model templated cases. There is a base templated type and an instantiated type extending from it.
+ */
+@scenarioService("/type/model/templated")
+namespace Type.Model.Templated;
+
+@friendlyName("{name}Type", T)
+model NumericType<T extends numeric> {
+  /**
+   * An array of numeric values.
+   */
+  values: T[];
+
+  value: T;
+}
+
+/**
+ * An instantiated type representing int32 type.
+ */
+model Int32Type extends NumericType<int32> {
+  /**
+   * The Kind of the Int32Type.
+   */
+  kind: "Int32Values";
+}
```

Comment:
it is really weird that this `Int32Type` extends from `NumericType` and no other properties added. could you try to change your model templated test to a more reasonable scenario.

Reply:
I guess `kind` is "added" in this model, though it is a constant.

Agree it is not very meaningful model. Do we have any source of such usage in service (and maybe adjust this a bit from that source)?

Reply:
Please see the story here https://github.com/Azure/cadl-ranch/pull/650#discussion_r1699342839

Reply:
Do you guys have suggestion on testing such case?

Reply:
I don't have any particular case. The thought is that `NumericValuesFacet<T extends numeric>` is not meaningful, if only `model Int32Type` extends it.

A more meaningful case is that there be another e.g. `model FloatType extends NumericValuesFacet<Float>`.

Reply:
I would prefer add another float. There are already a lot of "a direct use of xxx"  cases.

Reply:
A float type has been added